src/core/main.cpp

```diff
@@ -636,15 +666,13 @@ int main( int argc, char * * argv )
 		Engine::init( true );
 		destroyEngine = true;
 
-		QFileInfo fileInfo( fileToLoad );
-		if ( !fileInfo.exists() )
-		{
-			printf("The file %s does not exist!\n", fileToLoad.toStdString().c_str());
-			exit( 1 );
-		}
-
 		printf( "Loading project...\n" );
 		Engine::getSong()->loadProject( fileToLoad );
+		if( Engine::getSong()->isEmpty() )
+		{
+			printf("The project %s is empty, aborting!\n", fileToLoad.toUtf8().constData() );
+			exit( EXIT_FAILURE );
+		}
```

Comment:
You may omit `exit()` and avoid the export, so an upgraded `.lmmsrc.xml` can be written, but `exit()` (or `return`) is not a bad way.